.github/workflows/webui-periodic.yml.j2

```diff
@@ -56,54 +56,105 @@ jobs:
+            -v `pwd`/anaconda_rpms:/anaconda-rpms:ro \
+            -v `pwd`/images:/images:z \
+            --entrypoint /lorax-build-webui \
+            $ISO_BUILD_CONTAINER_NAME:$CONTAINER_TAG
+
+      - name: Clean up after lorax
+        if: always()
+        run: |
+          # remove container images together with the container
+          sudo podman rmi -f $ISO_BUILD_CONTAINER_NAME:$CONTAINER_TAG || true
+          sudo podman rmi -f $RPM_BUILD_CONTAINER_NAME:$CONTAINER_TAG || true
 
       - name: Create Permian settings file
         working-directory: ./permian
         run: |
           cat <<EOF > settings.ini
           [AnacondaWebUI]
+          anaconda_repo=file://${{ github.workspace }}/anaconda
           hypervisor_vm_limit=${{ env.TEST_JOBS }}
           test_timeout=${{ env.LAUNCHER_TIMEOUT_MINUTES }}
           [library]
-          directPath=${{ github.workspace }}/anaconda/ui/webui/test/integration
+          directPath=${{ github.workspace }}/anaconda/ui/webui/test/end2end
           EOF
 
-      - name: Run webui integration tests
+      - name: Run webui end2end tests
         working-directory: ./permian
         run: |
           PYTHONPATH=${PYTHONPATH:-}:${{ github.workspace }}/tplib \
           ./pipeline --debug-log permian.log \
             --settings settings.ini \
-            --override workflows.dry_run=False \
+            -o github.pull-request=${{ github.event.issue.number }} \
+            -o github.repository=${{ github.repository }} \
+            -o github.token=${{ secrets.GITHUB_TOKEN }} \
             run_event '{
-              "type":"github.scheduled.preview",
-              "InstallationSource":{
-                "base_repo_id": "bootiso",
-                "repos": {
-                  "bootiso":{
-                    "x86_64":{
-                      "os": "${{ steps.prepare_repo.outputs.repo_url }}",
-                      "kernel": "images/pxeboot/vmlinuz",
-                      "initrd": "images/pxeboot/initrd.img"
-                    }
-                  }
-                }
+              "type":"github.webui.e2e.scheduled",
+              "bootIso": {
+                "x86_64": "file://${{ github.workspace }}/anaconda/images/boot.iso"
               }
             }'
 
-      - name: Collect Permian logs
+      - name: Collect logs
         if: always()
         uses: actions/upload-artifact@v3
         with:
           name: 'logs'
           path: |
             permian/permian.log
             permian/local_logs/
+            permian/pipeline_data
+            permian/static/
+            permian/index.html
+            permian/xunit*.xml
             permian/*.dump
+
+      # Permian exit code doesn't reflect the test results
+      - name: Check tests results
+        if: always()
+        working-directory: ./permian
+        shell: bash {0}
+        run: |
+          fails=$( grep -E '<failure .* type="failure"/>|<error .* type="error"/>' xunit*.xml )
```

Comment:
Looks like this check should be improved. In your test it failed to catch that there are no xunit files https://github.com/rvykydal/anaconda/actions/runs/5877341654/job/15937287215
This applies to webui-tests on PR too.

Reply:
Updated (also a commit for the tests on PR).